### PR TITLE
test: ProductImageCarousel

### PR DESCRIPTION
**Pull request by sirugh**
## Description
Adds unit tests for Venia ProductImageCarousel.

## Related Issue

Related to #839.

## How Has This Been Tested?
`yarn run test`

[Image from Gyazo](https://gyazo.com/495355ed84947574c1ebd634517be130)

### CODE REVIEW

File: packages/venia-concept/src/components/ProductImageCarousel/carousel.js [hunk 1]
```diff
@@ -55,7 +55,7 @@ class Carousel extends Component {
             .sort((a, b) => {
                 const aPos = isNaN(a.position) ? 9999 : a.position;
                 const bPos = isNaN(b.position) ? 9999 : b.position;
-                return bPos - aPos;
```
Comment (sirugh) [inline packages/venia-concept/src/components/ProductImageCarousel/carousel.js]: @zetlen @jimbo I found in testing that this did the opposite of what I expected.

```js
var numbers = [4, 2, 5, 1, 3];
numbers.sort((a, b) => a - b); // [1, 2, 3, 4, 5]
numbers.sort((a, b) => b-a); // [5, 4, 3, 2, 1]
```

File: packages/venia-concept/src/components/ProductImageCarousel/thumbnailList.js [hunk 2]
```diff
@@ -19,12 +20,9 @@ class ThumbnailList extends Component {
                 file: PropTypes.string.isRequired
             })
         ).isRequired,
-        updateActiveItemIndex: PropTypes.func
+        updateActiveItemIndex: PropTypes.func.isRequired
     };
 
-    findItemIndexBySrc = src =>
-        this.props.items.findIndex(image => image.file === src);
-
```
Comment (supernova-at) [inline packages/venia-concept/src/components/ProductImageCarousel/thumbnailList.js]: 🎉

File: packages/venia-concept/src/components/ProductImageCarousel/thumbnailList.js [hunk 3]
```diff
@@ -19,12 +20,9 @@ class ThumbnailList extends Component {
                 file: PropTypes.string.isRequired
             })
         ).isRequired,
-        updateActiveItemIndex: PropTypes.func
+        updateActiveItemIndex: PropTypes.func.isRequired
```
Comment (supernova-at) [inline packages/venia-concept/src/components/ProductImageCarousel/thumbnailList.js]: 👍

**Comment by supernova-at** [comment]
Merge pending the resolution of https://github.com/magento-research/pwa-studio/pull/858#discussion_r254810474 .

**Pull request merged**